UIforETW/Utility.cpp

```diff
@@ -349,78 +558,83 @@ int DeleteFiles(HWND hwnd, const std::vector<std::wstring>& paths)
 		FO_DELETE,
 		&fileNames[0],
 		NULL,
-		FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION,
+		(FOF_ALLOWUNDO | FOF_FILESONLY | FOF_NOCONFIRMATION),
 	};
 	// Delete using the recycle bin.
-	int result = SHFileOperation(&fileOp);
-
-	return result;
+	//TODO: IFileOperation?
+	return ::SHFileOperationW(&fileOp);
 }
 
 void SetClipboardText(const std::wstring& text)
 {
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
+	ATLVERIFY(::OpenClipboard(::GetDesktopWindow()));
+	ATLVERIFY(::EmptyClipboard());
+
+	const size_t length = (text.size() + 1) * sizeof(text[0]);
+	HANDLE hmem = ::GlobalAlloc(GMEM_MOVEABLE, length);
+	if (!hmem)
+	{
+		ATLVERIFY(::CloseClipboard());
 		return;
+	}
 
-	EmptyClipboard();
+	void* const ptr = ::GlobalLock(hmem);
+	UIETWASSERT(ptr != NULL);
 
-	size_t length = (text.size() + 1) * sizeof(wchar_t);
-	HANDLE hmem = GlobalAlloc(GMEM_MOVEABLE, length);
-	if (hmem)
-	{
-		void *ptr = GlobalLock(hmem);
-		if (ptr != NULL)
-		{
-			memcpy(ptr, text.c_str(), length);
-			GlobalUnlock(hmem);
+	wcscpy_s(static_cast<wchar_t*>(ptr), (text.size() + 1), text.c_str());
 
-			SetClipboardData(CF_UNICODETEXT, hmem);
-		}
+	UnlockGlobalMemory(hmem);
+	if (::SetClipboardData(CF_UNICODETEXT, hmem) == NULL)
+	{
+		ATLVERIFY(!::GlobalFree(hmem));
+		ATLVERIFY(::CloseClipboard());
+		return;
 	}
-
-	CloseClipboard();
+	ATLVERIFY(::CloseClipboard());
 }
 
 std::wstring GetClipboardText()
 {
+	ATLVERIFY(::OpenClipboard(::GetDesktopWindow()));
 	std::wstring result;
-	BOOL cb = OpenClipboard(GetDesktopWindow());
-	if (!cb)
-		return result;
 
-	HANDLE hClip = GetClipboardData(CF_UNICODETEXT);
-	if (hClip)
+	HANDLE hClip = ::GetClipboardData(CF_UNICODETEXT);
+	UIETWASSERT(hClip);
+
+	void* const ptr = ::GlobalLock(hClip);
+	UIETWASSERT(ptr != NULL);
+	const size_t bytes = ::GlobalSize(hClip);
+	if (bytes == 0)
 	{
-		wchar_t *text = static_cast<wchar_t *>(GlobalLock(hClip));
-		if (text)
-		{
-			size_t bytes = GlobalSize(hClip);
-			result.insert(result.begin(), text, text + bytes / sizeof(wchar_t));
-		}
-		::GlobalUnlock(hClip);
+		UnlockGlobalMemory(hClip);
+		ATLVERIFY(::CloseClipboard());
+		return result;
 	}
-
-	CloseClipboard();
-
+	PCWSTR const text = static_cast<PCWSTR>(ptr);
+	result.insert(result.begin(), text, text + (bytes / sizeof(text[0])));
+	UnlockGlobalMemory(hClip);
+	ATLVERIFY(::CloseClipboard());
 	return result;
 }
 
 int64_t GetFileSize(const std::wstring& path)
 {
-	LARGE_INTEGER result;
-	HANDLE hFile = CreateFile(path.c_str(), GENERIC_READ,
-		FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING,
+	LARGE_INTEGER result = {0};
+	HANDLE hFile = ::CreateFileW(path.c_str(), GENERIC_READ,
+		(FILE_SHARE_READ | FILE_SHARE_WRITE), NULL, OPEN_EXISTING,
 		FILE_ATTRIBUTE_NORMAL, NULL);
+
 	if (hFile == INVALID_HANDLE_VALUE)
+	{
+		debugPrintf(L"Failed to get file size!\n");

```

Comment:
Remember that OutputDebugString prints globally, to any program (such as dbgview) that is listening. Therefore it is rude to print excessive messages. They may interfere with messages that a developer is trying to view from their own app.